src/DIRAC/Core/Utilities/Os.py

```diff
@@ -150,3 +152,25 @@ def sourceEnv(timeout, cmdTuple, inputEnv=None):
 @deprecated("Will be removed in DIRAC 8.1", onlyOnce=True)
 def which(executable):
     return shutil.which(executable)
+
+
+def findImage(operating_system="alma9"):
+    """Finds the image for the current platform"""
+    from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+
+    plat = DIRAC.gConfig.getValue("LocalSite/Platform", platform.machine())
+    if plat == "x86_64":
+        plat = "amd64"
+
+    CVMFS_locations = DIRAC.gConfig.getValue(
+        "LocalSite/CVMFS_locations", Operations().getValue("Pilot/CVMFS_locations", [])
+    )
+
+    rootImage = None
+
+    for candidate in CVMFS_locations:
+        rootImage = os.path.join(candidate, "containers", "os-base", f"{operating_system}-devel", "prod", plat)
+        if safe_listdir(rootImage):
```

Comment:
Is there something more specific than the existence of a folder to identify an image?

Reply:
Suggestions?

Reply:
`.singularity.d` ?

At least no every existing folder has that as a subfolder.

(@chrisburr ?)


```
ls -la /cvmfs/unpacked.cern.ch/gitlab-registry.cern.ch/sft/docker/alma9-core\:latest/
total 15
drwxr-xr-x. 19 cvmfs cvmfs 129 Mar 20 15:10 .
drwxr-xr-x. 70 cvmfs cvmfs  78 Jan 25  2021 ..
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar 25  2022 afs
lrwxrwxrwx.  1 cvmfs cvmfs   7 Mar  1 12:36 bin -> usr/bin
-rw-r--r--.  1 cvmfs cvmfs   0 Mar 20 15:10 .cvmfscatalog
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar  1 08:56 dev
lrwxrwxrwx.  1 cvmfs cvmfs  36 Mar 20 15:10 environment -> .singularity.d/env/90-environment.sh
drwxr-xr-x. 69 cvmfs cvmfs  38 Mar 20 15:10 etc
lrwxrwxrwx.  1 cvmfs cvmfs  27 Mar 20 15:10 .exec -> .singularity.d/actions/exec
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar 25  2022 home
lrwxrwxrwx.  1 cvmfs cvmfs   7 Mar  1 12:36 lib -> usr/lib
lrwxrwxrwx.  1 cvmfs cvmfs   9 Mar  1 12:36 lib64 -> usr/lib64
drwx------.  2 cvmfs cvmfs   6 Mar  1 08:56 lost+found
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar 25  2022 media
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar 25  2022 mnt
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar 25  2022 opt
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar  1 08:56 proc
drwxr-x---.  3 cvmfs cvmfs 149 Mar  1 12:36 root
drwxr-xr-x. 14 cvmfs cvmfs  22 Mar 20 15:09 run
lrwxrwxrwx.  1 cvmfs cvmfs  26 Mar 20 15:10 .run -> .singularity.d/actions/run
lrwxrwxrwx.  1 cvmfs cvmfs   8 Mar  1 12:36 sbin -> usr/sbin
lrwxrwxrwx.  1 cvmfs cvmfs  28 Mar 20 15:10 .shell -> .singularity.d/actions/shell
lrwxrwxrwx.  1 cvmfs cvmfs  24 Mar 20 15:10 singularity -> .singularity.d/runscript
drwxr-xr-x.  5 cvmfs cvmfs  80 Mar 20 15:10 .singularity.d
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar 25  2022 srv
drwxr-xr-x.  2 cvmfs cvmfs   6 Mar  1 08:56 sys
lrwxrwxrwx.  1 cvmfs cvmfs  27 Mar 20 15:10 .test -> .singularity.d/actions/test
drwxrwxrwx.  2 cvmfs cvmfs   6 Mar 20 14:55 tmp
drwxr-xr-x. 13 cvmfs cvmfs 107 Mar 20 15:09 usr
drwxr-xr-x. 19 cvmfs cvmfs  19 Mar 20 15:10 var
```

Reply:
I am not sure we want to rely on something called `singularity` when they changed the name to `apptainer`, it seems a bit of a moving target to me.

Reply:
Ahh, true. But I have no idea if there is anything that defines a container.
Thus pingin @chrisburr , any proposal, or is there no way?

Reply:
It’s not possible